### PR TITLE
Generate YDK model test cases

### PR DESCRIPTION
**Pull request by abhikeshav**
* Clean up api_model
 * Add ydk model tests to CI
 * Add subset of 0.4.2 profile models to generate model tests to run in CI
 * Some leaf types are not yet supported
 * CI test on model tests currently do not fail the CI run in case of failure/error

### CODE REVIEW

File: test/travis_ci.sh [hunk 1]
```diff
@@ -142,6 +157,15 @@ function init_confd {
     printf "\nInitializing confd\n"
 }
 
+function init_confd_model_test {
+    confd --stop
+    cp -f $YDKTEST_MODEL_FXS/* $CONFD_TARGET_DIR
```
Comment (psykokwak4) [inline test/travis_ci.sh]: I am thinking about split `CONFD_TARGET_DIR` for each different test cases. The `ydktest`, `ydk-model-test`, `deviation`, and `augmentation test` I am going to add will have their `CONFD_<test-case-name>_TARGET_DIR`, so the fxs files for different test cases will not mess up. Each `CONFD_<test-base-name>_TARGET_DIR` has same content except for their configuration file `confd.conf`(its relatively easy to modify confd.conf). Let me know what you think.

File: test/travis_ci.sh [hunk 2]
```diff
@@ -114,7 +100,7 @@ function teardown_env {
 }
 
 # compile yang to fxs
-function compile_yang_to_fxs {
+function compile_test_yang_to_fxs {
     rm -f $YDKTEST_FXS/*.fxs
     source $CONFD/confdrc
     cd $YDK_ROOT/yang/ydktest
```
Comment (psykokwak4) [inline test/travis_ci.sh]: `$YDK_ROOT` is same as `$YDKGEN_HOME`, maybe we could use one of them.

Comment (abhikeshav) [inline test/travis_ci.sh]: Agreed

File: test/travis_ci.sh (hunk 1 above)
Comment (abhikeshav) [inline test/travis_ci.sh]: Sure. Sounds good

**Pull request merged**